### ISSUE TITLE
unocss/cli is producing truncated output

### ISSUE THREAD

**Issue by daylinmorgan** [unocss/unocss#3930]
### UnoCSS version

v0.61.0

### Describe the bug

When running the `unocss` CLI with `html` and `js` from a Hugo theme the output file is both truncated and appears to be interspersing layers.

### Reproduction

You can see the resulting output when upgrading `@unocss/cli` from the last working version `v0.60.2` [here](https://github.com/daylinmorgan/brain-stem/compare/main...daylinmorgan:unocss-v0.61#diff-cd68c1265ad0ad210c0fcdd6d2817396b7b5903364f9f70fe35a24722d23f915).

The issue seems to have been introduced in `v0.60.3`. 

### System Info

```
> uname -a
Linux othalan 6.9.6 #1-NixOS SMP PREEMPT_DYNAMIC Fri Jun 21 12:40:40 UTC 2024 x86_64 GNU/Linux
```
```
> pnpm ls

devDependencies:
@catppuccin/palette 1.2.0
@iconify-json/openmoji 1.1.52
@unocss/cli 0.61.0
@unocss/reset 0.61.0
@unocss/transformer-directives 0.61.0
unocss 0.61.0
```

Invoking unocss with the following scripts:

```json
  "scripts": {
    "dev": "unocss \"layouts/**/*.html\" \"assets/js/*.js\" --watch -o assets/css/uno.css",
    "build": "unocss \"layouts/**/*.html\" \"assets/js/*.js\" -o assets/css/uno.css"
  },
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by wkeylin** [comment]
@Simon-He95 This issue was caused by this pull request #3803
https://github.com/unocss/unocss/blob/5d58d365fa5c664e65c385e8e91beaaad1be3407/packages/cli/src/index.ts#L167-L180
`Remove inline comments` results in an index going out of bounds in the layers array. As a result, some layers in the `cssCollection` have keys that are `undefined` and are not generated in the CSS. I have tried other methods, but issues arise during the deduplication process.
